src/wp-includes/nav-menu.php

```diff
@@ -780,6 +759,41 @@ function wp_get_nav_menu_items( $menu, $args = array() ) {
 	return apply_filters( 'wp_get_nav_menu_items', $items, $menu, $args );
 }
 
+/**
+ * Prime all linked objects to menu items.
+ *
+ * @param WP_Post[] $menu_items Array post objects of menu items.
+ */
+function _prime_menu_items_objects( $menu_items ) {
```

Comment:
* 🤔 This is consistent with the other priming functions but would be better as `_wp_prime...` -- I never know what is best in these situations in the absence of a time machine.
* Other priming functions accept an array of IDs, would that be an option here -- the use case is slightly different to the other functions of a similar name though.
* Should this function run the menu_item IDs through `_prime_post_caches()` to ensure they are in the cache -- it would be pretty ironic if attempting to prime one set of caches resulted in dozens of extra database calls.

Sorry, Jonny, I know I can be the worst sometimes :)

Reply:
I think is more similar to [update_term_cache](https://developer.wordpress.org/reference/functions/update_term_cache/), that accepts an array of terms. I could change this function accept an array of ids. Or at the very least, array_map an array of ids with get_post, so it accepts both ids and post objects.